### DISCUSSION TITLE
Drift guidance metrics

### DISCUSSION THREAD

**Discussion by rgrumbine** [dtcenter/METplus#2314] [Statistical Computation] [answered]
Though I start from a sea ice perspective, this is a general situation -- there is a feature whose location is known through time, whether a hurricane, storm system, sea ice floe, ... and we want to know how well the model forecast that motion.

In https://polar.ncep.noaa.gov/mmab/papers/tn315/ I discuss details of how to do the comparison and the relative performance of metrics as discriminators of forecast skill. Source codes in https://github.com/rgrumbine/ice_scoring/ (drift, Fortran and C++) and submodule SIDFEx (in R), in https://github.com/rgrumbine/ice_scoring/SIDFEx from the Sea Ice Dynamics Forecast Experiment -- https://sidfex.polarprediction.net/

Preferred measures are the directional bias and rms, the ratio between observed and modelled speed, and the error radius (distance between observed and forecast location).

The language of storm discussions is similar -- 'the model moved the storm too quickly', 'the model is veering the storm to the north of its real track' and such.

For the sea ice case, verification data are buoy locations from the IABP and IPAB (IO discussion already started)

**Comment by JohnHalleyGotway** [comment, marked as answer]
@rgrumbine thanks for providing links to those resources. As I mentioned in the METplus NOAA telecon this morning, yes, I clearly see the parallels with tropical cyclone verification. TC tracks are stored in ASCII files following the ATCF file format. And the MET TC tools, mainly tc_pairs and tc_stat, match up and compare the forecast and observed TC tracks.

I imagine many of the track error differences could be applied to those tracks as well. To address this in MET, there's several details we'd need to iron out.

I'll list some questions that come to mind but don't actually expect answers to them. I recommend that we identify a METplus scientist to collaborate more closely with you on these details.

- I realize we can use drifting buoys to define observation tracks. How do extract tracks from the models that are comparable to the buoys? Are we directly able to model the predicted drift of a buoy and compare it directly to the actual drift? Or are these not one-to-one comparisons, but instead bulk comparisons?
- Would need to decide how to store these "tracks".
- Would need to decide what MET tool should compare the forecast and observed tracks. Existing tool or perhaps a new one.
- Would need to decide what differences/statistics should be written and if/how they could be aggregated across multiple cases.

One example already existing in MET is the output from the MODE Time Domain tool. We have info to compare how forecast and observed object centroids move through time. Perhaps we could use that specific data to brainstorm how we'd analyze this type of data in a more generic way?

**Comment by rgrumbine** [comment]
Sorry, way too long since I checked here. I think the idea of a brainstorming session is the way to go. I have much of an answer in hand (code-wise), and another in hand (someone else's code, that's been exercised more broadly than mine). 

But much of the question is really about what would be the easiest path for migration in to Met+.

I realize (think) we're late in the cycle for a release, but would like to see it in the one after.

**Comment by j-opatz** [comment]
Hi @rgrumbine,

John HG mentioned in our stand up today that you were caught off-guard that this issue was closed. I wanted to take a moment to address this and provide an opportunity for you to comment if you had additional thoughts.

In our efforts to keep the METplus community engaged with Discussions and limit the amount of unanswered questions, we are trying to keep the amount of open Discussions to a minimum. This Discussion was opened on August 17, 2023, with the last comment made by yourself nearly a year later, stating a brainstorming session was the way to go for enhancing MET logic to track objects through time. While there are good ideas in this Discussion and worthwhile enhancements that could make their way into MET logic, the funding currently available to us does not allow us any time to sit down and collect these thoughts. With no way to move forward on the brainstorming request, no real issue to solve (more on that below), and nearly a year and a half since the last comment was made, I closed this Discussion out.

As mentioned, this Discussion doesn't really have an "issue" to solve; there are clearly capabilities you would like MET to have with regards to tracking drifting objects, and we can note those for ideas for future funding opportunities (we currently collect those ideas via a Github milestone of Backlog of Development Ideas). But keeping a Discussions open with no possible solution in sight is poor record keeping and not in the spirit of what these Discussions are meant for. 

Finally, it seems that there is a previous Discussions (dtcenter/METplus/Discussions/2312) that you opened the day before this one, that this Discussions seems related to. That prior discussion has much more activity and issues that John HG was able to investigate and provide comments (and issues) on. While that Discussion should also be revisited to see if we have solved (or at least addressed) all of the issues you raised, I see that Discussions as a more proper location to place this information. At least until such a time that we can dig deeper into these capabilities.

I've gone ahead and provided a comment on 2312 so these two Discussions are linked. We can also try and synthesize what exactly you'd like to see in MET that would address your concerns and place that into an issue (or multiple issues) for future consideration (if there isn't issues already detailing these ideas). But unless there is a current concern directly related to the topic that you listed in opening this Discussion that we can investigate and provide guidance on, I would like to have this Discussion closed by Friday April 3, 2026.

**Reply by rgrumbine** [reply]
It's a standing concern. I'm supposed to use Met for verification. Being able to verify the sea ice drift forecasts is important to me as a sea ice modeler. 

Met can't do that and there's no prospect of it doing so, you tell me now. If anyone complains about me not using Met, I'll point them here. I do have code to do such verification, now writing it in to my third language, python.

You may close the discussion.
